### PR TITLE
updated with minor adjustments to preprocessing and model params

### PR DESCRIPTION
**Pull request by zbuckley**

### CODE REVIEW

File: Code/CheXpertDataset.py [hunk 1]
```diff
@@ -66,14 +66,12 @@ def __call__(self, sample):
 class UZerosTransform:
     def __call__(self, sample):
         sample = sample[label_columns].replace(-1, 0)
-        sample = sample[label_columns].fillna(0)
         return sample
```
Comment (zbuckley) [inline Code/CheXpertDataset.py:69]: realized these lines aren't needed. see label_processing_uzeros and uones in the common_training file

File: Code/training_team8.py [hunk 2]
```diff
@@ -39,9 +40,10 @@
 data_loader_params = {
     'batch_size': BATCH_SIZE,
     'shuffle': True,  # why not.. *shrug*?
-    'num_workers': 12}
+    'num_workers': 16}
```
Comment (zbuckley) [inline Code/training_team8.py:43]: bumped this up for my 16 core machine.

File: Code/training_team8.py [hunk 3]
```diff
@@ -39,9 +40,10 @@
...
 
 training_loader, validation_loader = build_data_loaders(
+    base_label_transform,
```
Comment (zbuckley) [inline Code/training_team8.py:46]: added an ability to pass in the label transformer so that the training file controls it. 
Should better facilitate copying training file for each model we're training.

File: Code/training_common_utils.py [hunk 4]
```diff
@@ -141,10 +147,11 @@ def training_loop(
 
         model.train()
         # assuming loader will return image, label tuple
-        print("The type of training_loader is:",type(training_loader))
+        print("The type of training_loader is:", type(training_loader))
         for i, (images, labels) in enumerate(training_loader):
             print("\tMiniBatch {}:".format(i))
             minibatch_start = time.time()
+            #print("labels:", labels)
```
Comment (zbuckley) [inline Code/training_common_utils.py:155]: added this for debugging (was getting nan loss values again. 
then commented it out for more reasonable logging.

File: Code/training_common_utils.py [hunk 5]
```diff
@@ -187,11 +194,11 @@ def training_loop(
             loss_training_calc, loss_validation_calc))
         print("\tEpoch Compute Time: {}".format(
             time.time() - epoch_start))
-        if((loss_training_calc < min_train_loss) | (loss_validation_calc < min_val_loss):
+        if loss_training_calc < min_train_loss or loss_validation_calc < min_val_loss:
```
Comment (zbuckley) [inline Code/training_common_utils.py]: had to adjust the syntax. my python interpreter didn't like it as is.

Comment (zbuckley) [reply Code/training_common_utils.py]: as a thought... should it really be this instead? I think we may not be getting the desired behavior this way.
```suggestion
        if loss_validation_calc < min_val_loss:
```

File: Code/training_team8.py [hunk 6]
```diff
@@ -24,8 +25,8 @@
 # %% ----------------------------------- Hyper Parameters --------------------------------------------------------------
 LR = 5e-3
 MOMENT = .9
-BATCH_SIZE = 8
-N_EPOCHS = 50
+BATCH_SIZE = 16
+N_EPOCHS = 15
```
Comment (zbuckley) [inline Code/training_team8.py]: so far. on the full training set. I'm getting ~36 minutes per epoch. 
so 15 * 30 ~= 7.5 hours. If i'm right, I'll be done around 2/3am

File: Code/training_team8.py [hunk 7]
```diff
@@ -24,8 +25,8 @@
 # %% ----------------------------------- Hyper Parameters --------------------------------------------------------------
 LR = 5e-3
 MOMENT = .9
-BATCH_SIZE = 8
-N_EPOCHS = 50
+BATCH_SIZE = 16
```
Comment (zbuckley) [inline Code/training_team8.py:28]: increased this because my hardware supports it, and it'll speed things up. can be lowered if we run into CUDA ram errors.

File: Code/CheXpertDataset.py (hunk 1 above)
Comment (josh-bernstein618) [reply Code/CheXpertDataset.py:69]: Lol I meant to delete them from the other file when I changed this.  But since we're always changing blanks to 0s, this works too!

Comment (josh-bernstein618) [reply Code/CheXpertDataset.py:69]: Oh wait, you changed it in the other file.  Glad it wasn't an oversight on my part

Comment (zbuckley) [reply Code/CheXpertDataset.py:69]: I only changed it because for some reason it wasn't working before. don't remember why. but the NaN remover worked... so i simply composed them together... it also seems a bit cleaner.

File: Code/training_common_utils.py [hunk 8]
```diff
@@ -187,11 +195,11 @@ def training_loop(
             loss_training_calc, loss_validation_calc))
         print("\tEpoch Compute Time: {}".format(
             time.time() - epoch_start))
-        if((loss_training_calc < min_train_loss) | (loss_validation_calc < min_val_loss):
+        if loss_validation_calc < min_val_loss:
             print("*******Loss imrpoved*******")
```
Comment (josh-bernstein618) [inline Code/training_common_utils.py:199]:
```suggestion
            print("*******Loss improved*******")
```

File: Code/training_team8_uzeros.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
+# Starting from sample code:
```
Comment (josh-bernstein618) [inline Code/training_team8_uzeros.py:1]: I am not crazy about having 2 files that do almost the same thing.  The only difference is whether it imports `label_preprocessing_uzeros` or `label_preprocessing_uones` as base_label_transform.    I don't think this really justifies having 2 separate long files.  I am willing to accept this for our efficiency, but I wouldn't be surprised if Amir thinks it's ugly.

**Review by josh-bernstein618 — APPROVED** [review]
LGTM, did not test

File: Code/training_common_utils.py (hunk 5 above)
Comment (kshitijbhat330) [reply Code/training_common_utils.py]: Which version are you using? it works fine with my Python 3. Anyway, this is fine too

Comment (zbuckley) [reply Code/training_common_utils.py]: python 3.6.. 
i also realized later that we really shouldn't be basing it on loss_training_calc, since that is always  going to decrease (the whole backpropagation algorithm's convergence thing...) as the network "memorizes" the training data.

File: Code/training_team8_uzeros.py (hunk 9 above)
Comment (zbuckley) [reply Code/training_team8_uzeros.py:1]: Amir will not care. 
That being said my attempt to reduce copy-pasting was the common_training file. I'd intended to start copying the training files per model we need to train. If you want to go with another approach we'll need to set it up, but this should allow us to pursue separate models over the next few days without running into crazy merge conflicts.
